src/rexi/src/rexi_utils.erl

```diff
@@ -60,6 +60,16 @@ process_mailbox(RefList, Keypos, Fun, Acc0, TimeoutRef, PerMsgTO) ->
 
 process_message(RefList, Keypos, Fun, Acc0, TimeoutRef, PerMsgTO) ->
     receive
+        Msg ->
```

Comment:
It's worrying that we could be receiving  all the messages here? Maybe there is a low(no?) chance we'd get stray messages from other part of the code?